api/file_system.proto

```diff
@@ -0,0 +1,156 @@
+    // Restrictions:
+    // If given an absolute path, it needs to be under kubelet-csi-plugins-path
+    // or kubelet-pod-path - based on context, and be specified with drive letter
+    // prefix: e.g., "X:\".
+    // If given a relative path, it will be understood as being relative to either
+    // kubelet-csi-plugins-path or kubelet-pod-path - again based on context.
+    // Cannot already exist on host.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Maximum path length will be capped to 260 characters (MAX_PATH).
+    string path = 1;
+
+    // Context of the path creation used for path prefix validation
+    PathContext context = 2;
+}
+
+message MkdirResponse {
+    bool success = 1;
+
+    oneof error {
+        CmdletError cmdlet_error = 2;
+        string error_message = 3;
+    }
+}
+
+message RmdirRequest {
+    // The path to remove in the host filesystem
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    //
+    // Restrictions:
+    // If given an absolute path, it needs to be under kubelet-csi-plugins-path
+    // or kubelet-pod-path - based on context, and be specified with drive letter
+    // prefix: e.g., "X:\".
+    // If given a relative path, it will be understood as being relative to either
+    // kubelet-csi-plugins-path or kubelet-pod-path - again based on context.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Path cannot be a file of type symlink.
+    // Path needs to be a directory that is empty.
+    // Maximum path length will be capped to 260 characters (MAX_PATH).
+    string path = 1;
+
+    // Context of the path creation used for path prefix validation
+    PathContext context = 2;
+}
+
+message RmdirResponse {
+    bool success = 1;
+
+    oneof error {
+        CmdletError cmdlet_error = 2;
+        string error_message = 3;
+    }
+}
+
+message LinkPathRequest {
+    // Source of MkLink call to Windows
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    //
+    // Restrictions:
+    // If given an absolute path, it needs to be under kubelet-pod-path,
+    // and specified with drive letter prefix: "X:\".
+    // If given a relative path, it will be understood as being relative to
+    // kubelet-pod-path.
+    // Needs to already exist on host.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
```

Comment:
See my comment below and the [mount_windows package](https://github.com/kubernetes/kubernetes/blob/2036ee5ddeac235c0c39c705ad22cfa19bad309e/pkg/util/mount/mount_windows.go#L109) - I think symlinks with remote server shares as the source might be needed in order to mount SMB shares to containers.

Reply:
I think I'm missing something here, but as long as `MountSMBShareRequest` already symlinks the remote server share to within the proxy's working directories, why would users need to symlink to outside of it?

Again, apologies if I'm missing something :)